torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -288,7 +288,27 @@ def mlir_to_functions(op) -> None:
                 # child block to return the block result
                 return_ops = []
                 for block_id in child_block_ids:
-                    if name.startswith("scf."):
+                    if name == "scf.for":
+                        # example:
+                        # %result = scf.for %iv = %lb to %ub step %step iter_args(%arg = %init) -> (i32) ...
+                        # block args: 2 (%iv, %arg)
+                        # op operands: 4 (%lb, %ub, %step, %init)
+                        # `%arg` is mapping to `%init`
+                        for i, idx in enumerate(block_id_to_block_arg_ids[block_id]):
+                            if i == 0:
+                                next_fake_intermediate -= 1
+                                replacements[idx] = Intermediate(next_fake_intermediate)
+                            else:
+                                replacements[idx] = Intermediate(operand_ids[i+2])
```

Comment:
can you explain why there's a +2 here?

Reply:
operands[0] and operands[1] are lower-bound and upper-bound, where we don't have corresponding block args. The ^^ comments in code should give the example

Reply:
I think it refers to ```%init``` when ```i == 1```. I would replace the code with something like below:

```
nxt_fake_intermediate -= 1
idx = block_id_to_block_arg_ids[block_id][0]
replacements[idx] = Intermediate(next_fake_intermediate)
idx = block_id_to_block_arg_ids[block_id][1]
replacements[idx] = Intermediate(operand_ids[3])
```

Reply:
ohh, the num block args are not guaranteed to 2, it should be >= 1 based on the behavior